documentation/my-homeworks/week2/todo-app/src/App.js

```diff
@@ -0,0 +1,68 @@
+import React, { Component } from 'react'
+//import TodoList from './components/todo-list'
+import Container from './components/Container'
+
+
+const data = require('./todo.json');
+
+
+class App extends Component {
+  state = {
+    todoItems: []
+  }
+
+
+  componentWillMount() {
+    localStorage.getItem('todoItems') && this.setState({
+      todoItems: JSON.parse(localStorage.getItem('todoItems'))
+    })
+  }
+
+  componentDidMount() {
+    if (!localStorage.getItem('todoItems')) {
+      this.setState({
+        todoItems: data
+      })
+    } else {
+      console.log('we are using the data from the Local Storage.')
+    }
+
+  }
+  componentWillUpdate(nextProps, nextState) {
+    localStorage.setItem('todoItems', JSON.stringify(nextState.todoItems))
+  }
+
+  handelDoneStatus = (index) => {
+    const todoItems = JSON.parse(localStorage.getItem('todoItems'))
+    todoItems[index].done = !todoItems[index].done
+    this.setState({ todoItems })
+  }
+
+  handelNewTask = (newTask) => {
```

Comment:
Please use meaningful verbs for your functions and avoid using `handle` because it does not reflect the actual action of the function. You can replace it with `addNewTask`

Reply:
I can see that you are not using it anywhere also.